addons/binding/org.openhab.binding.draytonwiser/ESH-INF/thing/thing-types.xml

```diff
@@ -0,0 +1,429 @@
+                <required>true</required>
+            </parameter>
+            <parameter name="REFRESH" type="integer">
+                <label>Refresh interval</label>
+                <description>Interval in seconds between automatic refreshes</description>
+                <required>true</required>
+                <default>10</default>
+            </parameter>
+            <parameter name="AWAYSETPOINT" type="integer">
+	            <label>Away Mode Setpoint</label>
+	            <description>Set point temperature for away mode in degrees celsius</description>
+	            <required>true</required>
+	            <default>16</default>
+            </parameter>
+        </config-description>
+    </bridge-type>
+    
+    <!-- Controller Thing Type -->
+    <thing-type id="controller">
+        <supported-bridge-type-refs>
+            <bridge-type-ref id="heathub" />
+        </supported-bridge-type-refs>
+        
+        <label>Controller</label>
+        <description>The controller attached to the boiler</description>
+        
+        <channels>
+            <channel id="heatingOverride" typeId="heatingOverride-channel" />
+            <channel id="heatChannel1Demand" typeId="heatChannel1Demand-channel" />
+            <channel id="heatChannel1DemandState" typeId="heatChannel1DemandState-channel" />
+            <channel id="heatChannel2Demand" typeId="heatChannel2Demand-channel" />
+            <channel id="heatChannel2DemandState" typeId="heatChannel2DemandState-channel" />
+            <channel id="awayModeState" typeId="awayModeState-channel" />
+            <channel id="ecoModeState" typeId="ecoModeState-channel" />
+            <channel id="currentSignalRSSI" typeId="signalRSSI-channel" />
+            <channel id="currentSignalStrength" typeId="signalStrength-channel" />
+            <channel id="currentUnifiedSignalStrength" typeId="unifiedSignalStrength-channel" />
+        </channels>
+    </thing-type>
+    
+    <!-- Hot Water Thing Type -->
+    <thing-type id="hotwater">
+        <supported-bridge-type-refs>
+            <bridge-type-ref id="heathub" />
+        </supported-bridge-type-refs>
+        
+        <label>Hot Water</label>
+        <description>The hot water heating system</description>
+        
+        <channels>
+            <channel id="hotWaterOverride" typeId="hotWaterOverride-channel" />
+            <channel id="hotWaterDemandState" typeId="hotWaterDemandState-channel" />
+            <channel id="manualModeState" typeId="manualModeState-channel"/>
+            <channel id="hotWaterSetPoint" typeId="hotWaterSetPoint-channel"/>
+            <channel id="hotWaterBoostDuration" typeId="hotWaterBoostDuration-channel"/>
+            <channel id="hotWaterBoosted" typeId="hotWaterBoosted-channel"/>
+            <channel id="hotWaterBoostRemaining" typeId="hotWaterBoostRemaining-channel"/>
+        </channels>
+    </thing-type>
+    
+    <!-- Room Thing Type -->
+    <thing-type id="room">
```

Comment:
I terms of mapping the drayton devices to openHAB things I am wondering what the value of such a `room` thing is. As far as I understand there has to be a device (TRV or thermostat) inside the room to provide the values.

From what I can tell by a first look is that the room offers additional information about the windows which are certainly useful. The other channels seem to be the ones from the TRV or thermostats inside. Are they somehow aggregated if there are more than one TRV or thermostat inside the room (though more than one thermostat in a room doesn't make much sense)?

If you remove the TRV and thermostat channels from the room, it leaves the window and schedule information. I left a comment for the schedule channel already that we do not like to have such technical, plain JSON, channels.

To sum up all of the above, I would change the room to a `windowContact`, unless the aggregated data are of some value to the user. Also, how do these potentially aggregated values react on user changes to a TRV thing which they can directly manipulate? Wouldn't this become confusing for a user?

What is your opinion on this?

Reply:
The room concept makes sense to me.
The room temperature is an aggregation (well a min({...}) really) of all the sensor values assigned to that room (any combo of TRVs and thermostats).
There are operations that you perform on a room, such as boosting, or setting to manual mode (i.e. if there were multiple TRVs in a room, you wouldn't just boost one of them).
The other thing worth mentioning is that you don't directly assign setpoints etc to the TRVs, you assign them to the room. The only change you can make directly to a TRV thing is to set the locked/unlocked state (i.e. the child lock).